homeassistant/components/automation/__init__.py

```diff
@@ -588,7 +588,11 @@ async def _async_process_if(hass, config, p_config):
 
     def if_action(variables=None):
         """AND all conditions."""
-        return all(check(hass, variables) for check in checks)
+        try:
+            return all(check(hass, variables) for check in checks)
```

Comment:
Would it make sense to wrap each call to `check` with a `try…except` so we can say what condition had an error, ie "3rd condition" or maybe even allow condition alias.

Reply:
Yeah that makes sense but I agree it's better to explore such improvements in separate PRs.